experimental/dds/tree2/src/test/feature-libraries/sequence-field/sequenceChangeRebaser.spec.ts

```diff
@@ -346,6 +349,118 @@ describe("SequenceField - Rebaser Axioms", () => {
 	});
 });
 
+interface TestState {
+	length: number;
+	numNodes: number[];
+}
+
+type SequenceFieldTestState = FieldStateTree<TestState, TestChangeset>;
+
+/**
+ * See {@link ChildStateGenerator}
+ */
+const generateChildStates: ChildStateGenerator<TestState, TestChangeset> = function* (
+	state: SequenceFieldTestState,
+	tagFromIntention: (intention: number) => RevisionTag,
+	mintIntention: () => number,
+): Iterable<SequenceFieldTestState> {
+	const numNodes = state.content.numNodes;
+
+	// Undo the most recent edit
+	if (state.mostRecentEdit !== undefined) {
+		assert(state.parent?.content !== undefined, "Must have parent state to undo");
+		const undoIntention = mintIntention();
+		const invertedEdit = invert(state.mostRecentEdit.changeset);
+		yield {
+			content: state.parent.content,
+			mostRecentEdit: {
+				changeset: tagChange(invertedEdit, tagFromIntention(undoIntention)),
+				intention: undoIntention,
+				description: `Undo:${state.mostRecentEdit.description}`,
+			},
+			parent: state,
+		};
+	}
+
+	for (const nodeCount of numNodes) {
+		// MoveIn nodeCount nodes
+		const moveInIntention = mintIntention();
+		yield {
+			content: state.content,
+			mostRecentEdit: {
+				changeset: tagChange(
+					Change.move(1, nodeCount, 0),
+					tagFromIntention(moveInIntention),
+				),
+				intention: moveInIntention,
+				description: `MoveIn${nodeCount}${nodeCount === 1 ? "Node" : "Nodes"}From1To0`,
+			},
+			parent: state,
+		};
+
+		// MoveOut nodeCount nodes
+		const moveOutIntention = mintIntention();
+		yield {
+			content: state.content,
+			mostRecentEdit: {
+				changeset: tagChange(
+					Change.move(0, nodeCount, 1),
+					tagFromIntention(moveOutIntention),
+				),
+				intention: moveOutIntention,
+				description: `MoveOut${nodeCount}${nodeCount === 1 ? "Node" : "Nodes"}From0To1`,
+			},
+			parent: state,
+		};
+		for (let i = 0; i <= state.content.length; i++) {
+			// Insert nodeCount nodes
```

Comment:
I see you didn't include deletions. That's totally fine with me but since you've included everything else I just want to call it out in case that was not intentional.

Reply:
That was an unintentional oversight (I think I deleted them somewhere along the way refactoring). They should be included now.